fred/cogs/mediaonly.py

```diff
@@ -21,17 +36,41 @@ async def process_message(self, message: Message) -> bool:
             self.logger.info("Message doesn't contain media but the author is a T3", extra=common.message_info(message))
             return False
 
-        if config.MediaOnlyChannels.check(message.channel.id):
-            self.logger.info("Removing a message", extra=common.message_info(message))
-            await message.delete()
-            await self.bot.send_DM(
-                message.author,
-                f"Hi {message.author.name}, the channel you just tried to message in, "
-                f"{self.bot.get_channel(message.channel.id).name} is a 'Media Only' channel. "
-                f"This means you must attach a file in order to post there. "
-                f"Here is your message if you want to paste it again after the adequate changes: "
-                f"```\n{message.content}\n```",
-            )
-            return True
+        self.logger.info("Removing a message", extra=common.message_info(message))
+        await message.delete()
+        await self.bot.send_DM(
+            message.author,
+            f"Hi {message.author.mention}, "
+            f"{message.channel.mention} is a 'media-only' channel. "
+            f"This means posts there must have files or embedded links. "
+            f"Here is your message if you want to paste it again after the adequate changes: "
+            f"```\n{message.content}\n```",
+        )
+        return True
 
-        return False
+    async def _process_forum_message(self, message: Message) -> bool:
+
+        self.logger.info("Checking a thread", extra=common.message_info(message))
+
+        if len(message.embeds) > 0 or len(message.attachments) > 0:
+            self.logger.info("Message contains media", extra=common.message_info(message))
+            return False
+
+        ctx = await self.bot.get_context(message)
+
+        if await common.l4_only(ctx):
+            self.logger.info("Message doesn't contain media but the author is a T3", extra=common.message_info(message))
+            return False
+
+        self.logger.info("Removing a thread", extra=common.message_info(message))
+        await message.channel.delete()
+        await self.bot.send_DM(
+            message.author,
+            f"Hi {message.author.mention}, "
+            f"{message.channel.parent.mention} is a 'media-only' forum. "
+            f"This means posts there must have files or embedded links. "
+            f"Here is your message if you want to paste it again after the adequate changes: "
+            f"\n\n**Title: **`{message.channel.name}`\n"
+            f"```\n{message.content}\n```",
+        )
+        return True
```

Comment:
Please use a common function to not duplicate the checking logic. The function can format the message differently based on an argument or something similar

Reply:
Fixed in [ca002aa](https://github.com/satisfactorymodding/FICSIT-Fred/pull/145/commits/ca002aaf27da3b11041f78ce774ca2a9bd180cba)